### PR TITLE
Use bound parameters for the test function passed to LineClear

### PR DESCRIPTION
**Pull request by AJenbo**
Rebased https://github.com/diasurgical/devilutionX/pull/2396 cause i got excited

### CODE REVIEW

File: Source/controls/plrctrls.cpp [hunk 1]
```diff
@@ -91,7 +91,7 @@ int GetDistance(Point destination, int maxDistance)
 	}
 
 	int8_t walkpath[MAX_PATH_LENGTH];
-	int steps = FindPath(PosOkPlayer, MyPlayerId, Players[MyPlayerId].position.future.x, Players[MyPlayerId].position.future.y, destination.x, destination.y, walkpath);
+	int steps = FindPath(std::bind(PosOkPlayer, MyPlayerId, std::placeholders::_1), Players[MyPlayerId].position.future.x, Players[MyPlayerId].position.future.y, destination.x, destination.y, walkpath);
```
Comment (glebm) [inline Source/controls/plrctrls.cpp:94]: `std::bind` is basically obsolete and has issues. Use a lambda instead.
https://abseil.io/tips/108

Comment (AJenbo) [reply Source/controls/plrctrls.cpp:94]: To bad, I like the bind syntax better then the lambda syntax... but on the other hand I like lambda better as a concept...

Comment (AJenbo) [reply Source/controls/plrctrls.cpp:94]: Fixed via #2398, thanks for pointing it out.

Comment (AJenbo) [reply Source/controls/plrctrls.cpp:94]: Oh, nice using lambda also solves the issue with bind() seeing `IsTileAvailable(monster, Point)` being ambiguous so now I don't have to prefix it with `Monster` anymore :+1:

**Pull request merged**